django/db/backends/oracle/creation.py

```diff
@@ -40,6 +40,11 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
                 if keepdb:
                     return
                 sys.stderr.write("Got an error creating the test database: %s\n" % e)
+                if 'ORA-01031' in str(e):
+                    sys.stderr.write(
+                        "It appears the main user has insufficient privileges "
+                        "to create test database.\n")
+                    sys.exit(2)
```

Comment:
I think that this section ([creation.py#L38-L41](https://github.com/django/django/blob/master/django/db/backends/oracle/creation.py#L38-L41)):
```python
# if we want to keep the db, then no need to do any of the below,
# just return and skip it all.
if keepdb:
     return
```
should be removed (same as exception hiding in the test user creation https://github.com/django/django/commit/dacef9137f43fff88b527d1c02f6fe6a81e975aa), but rather in separate PR. 

As far as I'm concerned the only situation in which `_execute_test_db_creation` can raise exception that we want to omit (when `--keepdb` flag is `on`) is already checked in this [line](https://github.com/django/django/blob/master/django/db/backends/oracle/creation.py#L198) (`ORA-01543:  Tablespace "string" already exists`).